modules/uploads/scripts/list_for_user.ts

```diff
@@ -0,0 +1,38 @@
+import { ScriptContext } from "../_gen/scripts/list_for_user.ts";
+
+export interface Request {
+	userIds: string[];
+}
+
+export interface Response {
+	uploadIds: Record<string, string[]>;
+}
+
+export async function run(
+	ctx: ScriptContext,
+	req: Request,
+): Promise<Response> {
+	// Find uploads that match the IDs in the request
+	const dbUploads = await ctx.db.upload.findMany({
+		where: {
+			userId: {
+				in: req.userIds,
+			},
+			completedAt: { not: null },
+			deletedAt: null,
+		},
+		select: {
+			id: true,
+			userId: true,
+		},
+	});
+
+	// Map each userId to an array of upload IDs with that associated user ID
+	// TODO: There may be a more efficient way to do this? Not sure.
```

Comment:
You could probably iterate over the dbUploads and insert each into a `Map<UserId, Uploads[]>`. You'd have to benchmark both to see if theres even a difference though

Reply:
I actually did test that— it barely makes any difference.

Reply:
It may still be cleaner to do it the other way though.